internal/librarian/init.go

```diff
@@ -132,3 +139,72 @@ func setReleaseTrigger(state *config.LibrarianState, libraryID, libraryVersion s
 		library.ReleaseTriggered = trigger
 	}
 }
+
+// getLibraryChanges gets the changelog for the library with a non-empty libraryID,
+// if provided; or for all libraries if the libraryID is empty since the library
+// was last released.
+func getLibraryChanges(repo gitrepo.Repository, state *config.LibrarianState, libraryID string) error {
+	for i, library := range state.Libraries {
+		if libraryID != "" {
+			// Only get the changelog for one library.
+			if libraryID != library.ID {
+				continue
+			}
+			updatedLibrary, err := getChangesOf(repo, library)
+			if err != nil {
+				return err
+			}
+			state.Libraries[i] = updatedLibrary
+			break
+		}
+		// Get changelog for all libraries.
+		updatedLibrary, err := getChangesOf(repo, library)
+		if err != nil {
+			return err
+		}
+		state.Libraries[i] = updatedLibrary
+	}
+
+	return nil
+}
+
+// getChangesOf gets changelogs of the given library.
+func getChangesOf(repo gitrepo.Repository, library *config.LibraryState) (*config.LibraryState, error) {
+	commits, err := GetConventionalCommitsSinceLastRelease(repo, library)
+	if err != nil {
+		return nil, fmt.Errorf("failed to fetch conventional commits for library, %s: %w", library.ID, err)
+	}
+
+	changes := make([]*config.Change, 0)
+	for _, commit := range commits {
+		clNum := ""
+		if cl, ok := commit.Footers[KeyClNum]; ok {
+			clNum = cl
+		}
+
+		changeType := getChangeType(commit)
+		changes = append(changes, &config.Change{
+			Type:       changeType,
+			Subject:    commit.Description,
+			Body:       commit.Body,
+			ClNum:      clNum,
+			CommitHash: commit.SHA,
+		})
+	}
+
+	library.Changes = changes
+
+	return library, nil
+}
+
+// getChangeType gets the type of the commit, adding an escalation mark (!) if
+// it is a breaking change.
+func getChangeType(commit *gitrepo.ConventionalCommit) string {
+	changeType := commit.Type
+	_, ok := commit.Footers[KeyBreakingChange]
```

Comment:
I don't think you need this. `commit.IsBreaking` should already accounted for this footer. see https://github.com/googleapis/librarian/blob/7cb24f92fd31578a44aba491049e6c0f8242ad0b/internal/gitrepo/conventional_commits.go#L133-L135